src/background.js

```diff
@@ -193,43 +194,102 @@ const rollout = {
     await browser.storage.local.set({[name]: value});
   },
 
-  async trrModePrefHasUserValueCheck() {
-    log("trrModePrefHasUserValueCheck");
+  async decodeTrrMode(number){
+    // This turns whatever trr.mode pref the user set into either enable_doh or disable_doh.
+    let dohStatus;
+    switch (number) {
+    case 0:
+      dohStatus = "disable_doh";
+      break;
+    case 2:
+      dohStatus = "enable_doh";
+      break;
+    case 3:
+      dohStatus = "enable_doh";
+      break;
+    case 5:
+      dohStatus = "disable_doh";
+      break;
+    }
+    return dohStatus;
+  },
+
+  async trrModePrefHasUserValueAndEnterprisePolicyCheck(event) {
+    // Cache heuristics info in case a ping about a policy discovery event is sent
+    let results = await runHeuristics();
+    results.evaluateReason = event;
+
+    // Reset skipHeuristicsCheck
+    this.setSetting("skipHeuristicsCheck", false);
+
     // This confirms if a user has modified DoH (via the TRR_MODE_PREF) outside of the addon
-    // This runs only on the FIRST time that add-on is enabled, and if the stored pref
-    // mismatches the current pref (Meaning something outside of the add-on has changed it
+    // This runs only on the FIRST time that add-on is enabled and if the stored pref
+    // mismatches the current pref (Meaning something outside of the add-on has changed it)
+
     if (
       await browser.experiments.preferences.prefHasUserValue(
         TRR_MODE_PREF)
     ) {
-      // TODO: Add telemtery ping for user who already has pref on DoH
+      let curMode = await browser.experiments.preferences.getUserPref(TRR_MODE_PREF, 0);
+      curMode = await this.decodeTrrMode(curMode);
```

Comment:
It would be clearer to set this to it's own var. Can we make the response value functional code in disabling doh?

Reply:
For example it seems we only use this function to send to telemetry, whatever we use elsewhere might get out of date or subtly different.